app/controllers/works_controller.rb

```diff
@@ -0,0 +1,90 @@
+class WorksController < ApplicationController
+  def index
+    @books = Work.sort_by_category("book")
+    @movies = Work.sort_by_category("movie")
+    @albums = Work.sort_by_category("album")
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def edit
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
+      return
+    elsif @work.update(work_params)
+      redirect_to work_path(@work.id)
+      return
+    else
+      render :edit
```

Comment:
It's best to include a response code for all renders, and also use `flash.now` to present validation errors to the user.

```suggestion
      flash.now[:error] = @work.errors.messages
      render :edit, status: :bad_request
```